pkg/network/ebpf/c/prebuilt/usm.c

```diff
@@ -338,7 +359,9 @@ SEC("uprobe/gnutls_handshake")
 int uprobe__gnutls_handshake(struct pt_regs* ctx) {
     u64 pid_tgid = bpf_get_current_pid_tgid();
     void *ssl_ctx = (void *)PT_REGS_PARM1(ctx);
-    bpf_map_update_with_telemetry(ssl_ctx_by_pid_tgid, &pid_tgid, &ssl_ctx, BPF_ANY);
+    ssl_handshake_args_t args = { 0 };
+    args.ctx = ssl_ctx;
+    bpf_map_update_with_telemetry(ssl_ctx_by_pid_tgid, &pid_tgid, &args, BPF_ANY);
```

Comment:
does every library supports the async mode? isn't it just netty?

Reply:
don't you need to mimic the same implementation you added for `uretprobe__SSL_do_handshake` to `uretprobe__gnutls_handshake` as wekk?